File: schemaorg/sportsorganization.go
package schemaorg

import "encoding/json"

// SportsOrganization see : https://schema.org/SportsOrganization
type SportsOrganization struct {

typeContext

Organization

// Sport see : https://schema.org/sport
// A type of sport (e.g. Baseball).
Sport interface{} `json:"sport"` // types : Text URL

}

func (v *SportsOrganization) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "SportsOrganization"

	return json.Marshal(v)
}

File: schemaorgphp/classes/sports_organization.php
<?php

class SportsOrganization extends Organization implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'SportsOrganization';
	
	/**
	 * A type of sport (e.g. Baseball).
	 * see : https://schema.org/sport
	 * @var string|string[]|string|string[]
	 */
	public var $sport;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'SportsOrganization'
		);
		
		$serialized = so_json_serialize( $this->sport );
		if ( ! empty( $serialized ) ) {
			$out['sport'] = $serialized;
		}
		
		return $out;
	}
}
